### PR TITLE
Feat: OPX3.1-dev1

### PR DESCRIPTION
**Pull request by GarrickHe**
* Feature: Handle new VXLAN SAI attributes
* Update: Display TCAM utilization
* Update: Add egress filtering attribute

Signed-off-by: Garrick He <garrick_he@dell.com>

### CODE REVIEW

File: opx/sai_port_common.h [hunk 1]
```diff
@@ -319,7 +319,6 @@ typedef struct _sai_port_info_t
 
     /** Number of VLANs that the port is part of */
     uint_t                num_vlans;
-
```
Comment (atanu-mandal) [inline opx/sai_port_common.h]: No such change. Revert ?

File: sos/Makefile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+SOSREPORT_PLUGINS = plugins/dn_sai_common.py
```
Comment (atanu-mandal) [inline sos/Makefile]: Please don't bring in sos files.

File: sos/plugins/dn_sai_common.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from sos.plugins import Plugin, DebianPlugin
```
Comment (atanu-mandal) [inline sos/plugins/dn_sai_common.py]: Please don't bring in sos files.

File: src/acl/sai_acl_slice.c [hunk 4]
```diff
@@ -0,0 +1,242 @@
+/************************************************************************
+ * * LEGALESE:   "Copyright (c) 2017, Dell Inc. All rights reserved."
```
Comment (atanu-mandal) [inline src/acl/sai_acl_slice.c]: Copyright update?

File: src/acl/sai_vm_acl_slice.c [hunk 5]
```diff
@@ -0,0 +1,172 @@
+/************************************************************************
+ * * LEGALESE:   "Copyright (c) 2015, Dell Inc. All rights reserved."
```
Comment (atanu-mandal) [inline src/acl/sai_vm_acl_slice.c]: Copyright update?

File: src/fc/sai_vm_fc_switch.c [hunk 6]
```diff
@@ -73,3 +72,4 @@ sai_fc_switch_api_t *sai_fc_switch_api_query()
 
     return (&sai_fc_switch_method_table);
 }
+
```
Comment (atanu-mandal) [inline src/fc/sai_vm_fc_switch.c]: No such changes. Revert ?

File: src/unit_test/run_test [hunk 7]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+#
+# Copyright (c) 2017 Dell Inc.
```
Comment (atanu-mandal) [inline src/unit_test/run_test]: Copyright update?

File: opx/sai_port_common.h (hunk 1 above)
Comment (GarrickHe) [reply opx/sai_port_common.h]: No. this is fine.

File: src/acl/sai_acl_slice.c (hunk 4 above)
Comment (GarrickHe) [reply src/acl/sai_acl_slice.c]: will update with script.

**Comment by GarrickHe** [comment]
Pull-request updated, HEAD is now e03aa99e633bf43bc1ffd197f038a8ec6b67d722

**Pull request merged**